plugin/main/src/kotlinx/benchmark/gradle/BenchmarksPlugin.kt

```diff
@@ -1,10 +1,14 @@
 package kotlinx.benchmark.gradle
 
+import kotlinx.benchmark.gradle.internal.KotlinxBenchmarkPluginInternalApi
 import org.gradle.api.*
 import org.gradle.util.*
 
 @Suppress("unused")
-class BenchmarksPlugin : Plugin<Project> {
+abstract class BenchmarksPlugin
+@KotlinxBenchmarkPluginInternalApi
+constructor() : Plugin<Project> {
+
     companion object {
```

Comment:
I think it's best to keep it public. Are there any constants in particular need that should be hidden?

It's quite common for plugins to expose string constants that are used for naming. It can help when configuring conventions in a binary plugin, where the DSL accessors aren't present.

For example:

```kotlin
abstract class FooPlugin : Plugin<Project> {

   override fun apply(project: Project)
        project.pluginManager.apply("org.jetbrains.kotlinx.benchmark")

        val someCustomTask = project.tasks.create("...") {
            it.dependsOn(
                project.tasks.named(BenchmarksPlugin.RUN_BENCHMARKS_TASKNAME)
            )
        }
   }
}
```

Reply:
Understood. In that case, let's hide only `*_SUFFIX` and `*_DEPENDENCY` constants. Or do you think they could also useful?

Reply:
Sure, good idea, I'll add that.

BTW, I see some of these changes as a precursor to changing the visibility to `internal` in a future release. This would make more sense, and would be 'cleaner' in the code. What do you think?

Reply:
Hmm, it looks like `const val` properties can't be hidden from the API dump https://github.com/Kotlin/dokka/pull/3549

There are two options:

1. Annotate them with `@BenchmarkInternal`, but they'll still appear the in the API dump.

2. Mark the existing properties as `@Deprecated`, but make copies for internal usage inside of a new `internal object`, or a new file `constants.kt` and marked as `internal`?
    
    Example:
    
    ```kotlin
        companion {
            @Deprecated("Internal Benchmark constant - please replace with a string value")
            const val BENCHMARK_GENERATE_SUFFIX = Constants.BENCHMARK_GENERATE_SUFFIX
            // ...
        }
    
        internal object Constants {
            const val BENCHMARK_GENERATE_SUFFIX = "BenchmarkGenerate"
            // ...
        }
    ```

What are your thoughts?

Reply:
I am fine with annotating each of the constants we want to hide.

Reply:
Done!

P.S. I linked to the wrong issue above, I meant to link https://github.com/Kotlin/binary-compatibility-validator/issues/90